lib/action_view/component/base.rb

```diff
@@ -189,20 +190,32 @@ def templates
             end
         end
 
-        def validate_templates
+        def valid?(raise_if_invalid)
```

Comment:
Not a common practice to have question mark methods triggering exceptions. I think this should be renamed to something like `validate`.

Reply:
Agreed. We don't really _need_ to have this method have a return value, as it halts the call chain by raising an error if there are any issues.

Reply:
we do need it to return a value because we need the ability to call compile, not raise an exception but not blow up later because templates are screwy. I'll figure out a solution